File: chishibekabwe7-learning-Repository/Data Structures & Algorithms with Java Programming/C++ Programming/Stack Data Structure.cpp
#include <iostream>
#include <stack>
using namespace std;

int main(){
    //Declaring a Stack Data Structure
    stack<int> MyStack_Chishibe;

    //Adding elements to out Stack
    MyStack_Chishibe.push(10);
    MyStack_Chishibe.push(20);
    MyStack_Chishibe.push(30);
    MyStack_Chishibe.push(40);

    //Removing elements from our Stack
    MyStack_Chishibe.pop();  //Removes the top element

    //Accesing the elements with a variable
    int topElement = MyStack_Chishibe.top();
    cout<<"The top element is "<<topElement<<" in Chishibe's Stack"<<endl;

    //Checking if the stack is empty
    if(MyStack_Chishibe.empty()){
        cout<<"Chishibe's stack is empty"<<endl;
    }
    else{
        cout<<"Chishibe's stack is not empty"<<endl;
    }

    //Checking the size of the stack
    int stackSize = MyStack_Chishibe.size();
    cout<<"Chishibe's stack size is "<<stackSize<<endl;

    return 0;
}

File: chishibekabwe7-learning-Repository/chishibekabwe7-learning-Repository/Data Structures & Algorithms with Java Programming/Python Data Structures/Stack Data Structure.py
#!/usr/bin/env python3

#Declaring an empty stack
my_stack = []

#Pushing elements onto the stack
my_stack.append(20)
my_stack.append("Chishibe Kabwe")
my_stack.append("Male")
my_stack.append("Digital Creater")

#Poping elements from stack
Removed_element = my_stack.pop()
print(f"The Poped element was {Removed_element}!")


#Checking if the stack is empty
if my_stack:
    print("Stack is not empty")
else:
    print("Stack is empty")

#Checking the size of the stack with a variable
stack_size = len(my_stack)
print(f"The size of the array is {stack_size}")

#Accessing the top element 
top_element = my_stack[-1]
print(f"The top element of my stack is {top_element}")




